test/integration/match2.test.ts

```diff
@@ -141,17 +158,92 @@ describe('match2', () => {
     it('it should get all proposal transactions - 200', async () => {
       const response = await get(app, `/match2/${seededMatch2Id}/proposal`)
       expect(response.status).to.equal(200)
-      expect(response.body).to.deep.equal([
-        {
-          id: seededProposalTransactionId,
-          apiType: TransactionApiType.match2,
-          transactionType: TransactionType.proposal,
-          localId: seededMatch2Id,
-          state: TransactionState.submitted,
-          submittedAt: exampleDate,
-          updatedAt: exampleDate,
-        },
-      ])
+      expect(response.body.length).to.be.greaterThan(0)
+      expect(response.body[0]).to.deep.equal({
+        id: seededProposalTransactionId,
+        transactionType: TransactionType.proposal,
+        apiType: TransactionApiType.match2,
+        localId: seededMatch2Id,
+        state: TransactionState.submitted,
+        submittedAt: exampleDate,
+        updatedAt: exampleDate,
+      })
+    })
+
+    it('should acceptA then acceptFinal a match2 on-chain', async () => {
+      match2AcceptMock()
+      // submit to chain
+      const responseAcceptA = await post(app, `/match2/${seededMatch2Id}/accept`, {})
+      expect(responseAcceptA.status).to.equal(201)
+
+      // check local entities update with token id
+      const [match2AcceptA] = await db.getMatch2(seededMatch2Id)
+      expect(match2AcceptA.latestTokenId).to.equal(match2AcceptMockTokenId)
+      expect(match2AcceptA.state).to.equal(Match2State.acceptedA)
+      expect(match2AcceptA.originalTokenId).to.equal(seededMatch2TokenId)
+
+      match2AcceptFinalMock()
+      // submit to chain
+      const responseAcceptFinal = await post(app, `/match2/${seededMatch2Id}/accept`, {})
+      expect(responseAcceptFinal.status).to.equal(201)
+
+      // check local entities update with token id
+      const [demandA] = await db.getDemand(seededOrderId)
+      expect(demandA.latestTokenId).to.equal(match2AcceptFinalMockTokenIds[0])
+      expect(demandA.state).to.equal(DemandState.allocated)
+      expect(demandA.originalTokenId).to.equal(seededOrderTokenId)
+
+      const [demandB] = await db.getDemand(seededCapacityId)
+      expect(demandB.latestTokenId).to.equal(match2AcceptFinalMockTokenIds[1])
+      expect(demandB.state).to.equal(DemandState.allocated)
+      expect(demandB.originalTokenId).to.equal(seededCapacityTokenId)
+
+      const [matchAcceptFinal] = await db.getMatch2(seededMatch2Id)
+      expect(matchAcceptFinal.latestTokenId).to.equal(match2AcceptFinalMockTokenIds[2])
+      expect(matchAcceptFinal.state).to.equal(Match2State.acceptedFinal)
+      expect(matchAcceptFinal.originalTokenId).to.equal(seededMatch2TokenId)
+    })
```

Comment:
I won't reject for this but these tests are getting a bit unweildy. We should look to getting a more structured:

1. setup
2. assertion
3. teardown

Methodology where we're testing one thing clearly at a time. This is fine for now but we're on that cusp